tests/0151-purge-brokers.c

```diff
@@ -0,0 +1,410 @@
+            {TEST_ACTION_ADD_BROKER, 6},
+        };
+
+        do_test_add_remove_brokers(3, actions, expected_broker_ids,
+                                   RD_ARRAY_SIZE(expected_broker_ids),
+                                   expected_brokers_cnt, NULL, NULL);
+
+        SUB_TEST_PASS();
+}
+
+/**
+ * @brief Test setting down all brokers from the mock cluster,
+ *        simulating a correct cluster roll that never sets down the majority
+ *        of brokers.
+ *
+ *        The effect is similar to decommissioning the brokers. Partition
+ *        reassignment is not triggered in this case but they are not announced
+ *        anymore by the Metadata response.
+ */
+static void do_test_cluster_roll(void) {
+        SUB_TEST_QUICK();
+
+        int32_t expected_brokers_cnt[] = {5, 5, 4, 3, 4, 3, 4,
+                                          3, 4, 4, 3, 4, 5};
+
+        int32_t expected_broker_ids[][5] = {
+            {1, 2, 3, 4, 5}, {1, 2, 3, 4, 5}, {2, 3, 4, 5}, {3, 4, 5},
+            {1, 3, 4, 5},    {1, 4, 5},       {1, 2, 4, 5}, {1, 2, 5},
+            {1, 2, 3, 5},    {1, 2, 3, 5},    {1, 2, 3},    {1, 2, 3, 4},
+            {1, 2, 3, 4, 5}};
+
+        int32_t actions[][2] = {
+            {TEST_ACTION_SET_GROUP_COORDINATOR, 5},
+            {TEST_ACTION_SET_DOWN_BROKER, 1},
+            {TEST_ACTION_SET_DOWN_BROKER, 2},
+            {TEST_ACTION_SET_UP_BROKER, 1},
+            {TEST_ACTION_SET_DOWN_BROKER, 3},
+            {TEST_ACTION_SET_UP_BROKER, 2},
+            {TEST_ACTION_SET_DOWN_BROKER, 4},
+            {TEST_ACTION_SET_UP_BROKER, 3},
+            {TEST_ACTION_SET_GROUP_COORDINATOR, 1},
+            {TEST_ACTION_SET_DOWN_BROKER, 5},
+            {TEST_ACTION_SET_UP_BROKER, 4},
+            {TEST_ACTION_SET_UP_BROKER, 5},
+        };
+
+        do_test_add_remove_brokers(5, actions, expected_broker_ids,
+                                   RD_ARRAY_SIZE(expected_broker_ids),
+                                   expected_brokers_cnt, NULL, NULL);
+
+        SUB_TEST_PASS();
+}
+
+/**
+ * @brief Log callback that waits for the TERMINATE op to be received
+ */
+static void do_test_remove_then_add_log_cb(const rd_kafka_t *rk,
+                                           int level,
+                                           const char *fac,
+                                           const char *buf) {
+        if (!rd_atomic32_get(&do_test_remove_then_add_received_terminate) &&
+            strstr(buf, "/1: Handle terminates in state") > 0) {
+                rd_atomic32_set(&do_test_remove_then_add_received_terminate, 1);
+                while (!rd_atomic32_get(&verification_complete))
+                        rd_usleep(100 * 1000, 0);
+        }
+}
+
+/**
+ * @brief Await for the TERMINATE op to be received after the action
+ *        that removes the broker then proceed to
+ *        add the broker again.
+ */
+static rd_bool_t do_test_remove_then_add_await_after_action_cb(int action) {
+        if (action == 2) {
```

Comment:
```suggestion
        if (action == TEST_ACTION_SET_DOWN_BROKER) {
```
Is this what you mean?

Reply:
No, here it's the action number that is passed (starting from 1) not the action type